lib/graph_records.rb

```diff
@@ -86,49 +303,87 @@ def inspect(obj)
         @nodes << it
         @edges << order_nodes(obj, it)
 
-        inspect(it)
+        if @nodes.length > @node_limit
+          raise "Recursion limit of #{@node_limit} nodes has been exceeded. Try restricting the graph."
+        end
+
+        introspect(it)
       end
     end
   end
 
   def get_associated_objects(obj, association_name)
     obj
       .send(association_name)
-      .then do |associated_objects|
-        load_association(association_name, associated_objects)
+      .then { |associated_objects| load_association(associated_objects) }
+  end
+
+  def load_association(associated_objects)
+    Array(
+      if associated_objects.is_a?(ActiveRecord::Relation)
+        associated_objects.strict_loading!(false)
+      else
+        associated_objects
       end
+    )
   end
 
-  def load_association(association_name, associated_objects)
-    if respond_to?("#{association_name}_loader")
-      send("#{association_name}_loader", associated_objects)
-    else
-      associated_objects
+  def order_nodes(*nodes)
+    nodes.sort_by do |node|
+      @node_order.index(node.class.name.underscore.to_sym) || Float::INFINITY
```

Comment:
If we were to change `DEFAULT_NODE_ORDER` to be strings instead of symbols, this could be slightly simplified to:

```suggestion
      @node_order.index(node.class.name.underscore) || Float::INFINITY
```